packages/react-components/react-tree/src/components/TreeItemLayout/useTreeItemLayout.ts

```diff
@@ -32,6 +34,8 @@ export const useTreeItemLayout_unstable = (
       iconAfter: 'div',
       actions: 'div',
       aside: 'div',
+      // Casting here to a union between checkbox and radio
+      selector: (selectionMode === 'multiselect' ? Checkbox : Radio) as React.ElementType<CheckboxProps | RadioProps>,
```

Comment:
if the selection indicator is rendered on the TreeItemLayout, why bother passing it from the TreeItem through context?

Reply:
the same goes for `actions`, `aside` and `expandIcon`. Although they're rendered inside the `*Layout` components, their declaration in the `TreeItem` itself seems more logical, as they're linked with the `TreeItem` logic meanwhile layout component is only there to present things in the proper design position, therefore it should only worry about presentation slots, like `iconBefore` and `iconAfter`, which don't hold much logic

Reply:
so if a users uses those slots without the layout component, they are useless?

Reply:
Let's have this discussion in a separate PR 👍🏻